packages/react/src/context.js

```diff
@@ -11,23 +11,28 @@ export const withMDXComponents = Component => props => {
 
 export const useMDXComponents = components => {
   const contextComponents = React.useContext(MDXContext)
-  let allComponents = contextComponents
-  if (components) {
-    allComponents = isFunction(components)
-      ? components(contextComponents)
-      : {...contextComponents, ...components}
+
+  // Custom merge via a function prop
+  if (isFunction(components)) {
+    return components(contextComponents)
   }
 
-  return allComponents
+  return {...contextComponents, ...components}
 }
 
-export const MDXProvider = props => {
-  const allComponents = useMDXComponents(props.components)
+export const MDXProvider = ({
+  components,
+  children,
+  useParentContext = true
```

Comment:
:thinking: 
JSX/HTML generally makes the absences of an attribute false, and the presence true.
Here the absence is true, the presence is also true, and only an explicit `false` will disable it.

This could both cause issues with adopters that use https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/jsx-boolean-value.md and be not intuitive for adopters to use.

Maybe `disableParentContext` would make sense?
Thoughts? :thought_balloon:

Reply:
Yeah, that's a fair point. A `disable` style attr would be more ergonomic because that's what a user is _intending_. Will update!